Analyzer: Do not override abstract methods with missing methods

gzm0 opened the pull request.

sjrd reviewed (commented):
Looks good, but the new test does not pass on the CI.

gzm0 commented:
Updated. I was confused again between `lookupStaticLikeMethod` (which cannot happen for Public methods) and `reachStatic`, which can happen.

I've added an additional test so that we test both `reach()` and `reachStatic()` on abstract methods.
